kfs-core/src/main/java/edu/arizona/kfs/fp/service/impl/PaymentMethodGeneralLedgerPendingEntryServiceImpl.java

```diff
@@ -0,0 +1,455 @@
+                
+                // Exclude use tax PREQ/CM with AP PCard and Use Tax
+                if ((apOffsetEntry.getFinancialDocumentTypeCode().equals(PRNC) || apOffsetEntry.getFinancialDocumentTypeCode().equals(CMNC)) 
+                        && pm.getPaymentMethodCode().equals(AP_CREDIT_CARD_PAYMENT_METHOD_CODE)) {
+                	if (getPurchasingAccountsPayableModuleService().hasUseTax(document.getDocumentHeader().getDocumentNumber())) {
+                        KualiDecimal totalPreTaxDollarAmount = getPurchasingAccountsPayableModuleService().getTotalPreTaxDollarAmount(document.getDocumentHeader().getDocumentNumber());
+                        apOffsetEntry.setTransactionLedgerEntryAmount(totalPreTaxDollarAmount);
+                	}
+                }
+                   
+                document.addPendingEntry(apOffsetEntry);
+                sequenceHelper.increment();
+                
+                // handle the offset entry
+                GeneralLedgerPendingEntry offsetEntry = new GeneralLedgerPendingEntry(apOffsetEntry);
+                getGeneralLedgerPendingEntryService().populateOffsetGeneralLedgerPendingEntry(document.getPostingYear(), apOffsetEntry, sequenceHelper, offsetEntry);
+                    
+                document.addPendingEntry(offsetEntry);
+                sequenceHelper.increment();
+            }
+        }
+        
+        return true;
+    }
+
+    /**
+     * If bank specification is enabled generates bank offsetting entries for the document amount
+     * 
+     */
+    public boolean generateDocumentBankOffsetEntries(AccountingDocument document, String bankCode, String bankCodePropertyName, String documentTypeCode, GeneralLedgerPendingEntrySequenceHelper sequenceHelper, KualiDecimal bankOffsetAmount, boolean reverseCharge ) {
+        boolean success = true;
+
+        if (!SpringContext.getBean(BankService.class).isBankSpecificationEnabled()) {
+            return success;
+        }
+        
+        // Exclude use tax PREQ with AP PCard and Use Tax
+        // Overage amounts are posting to the wrong account
+        // After a similar problem was experienced with Unit Price Overage corrections, it was
+        // decided that a PRNC with bank 0007 (AP Credit Card) should never create bank offsets 
+        if (documentTypeCode.equals(PRNC) && bankCode.equals(AP_CREDIT_CARD_BANK_CODE)) {
+        	return success;
+        }
+        
+        Bank bank = getBankService().getByPrimaryId(bankCode);
+
+        if ( bankOffsetAmount == null ) {
+            // When it's a use tax PRNC or CMNC, create the bank offset for the total pre tax amount.
+            if(PRNC.equals(documentTypeCode) || CMNC.equals(documentTypeCode)) {
+	            if (getPurchasingAccountsPayableModuleService().hasUseTax(document.getDocumentHeader().getDocumentNumber())) {
+	            	if(PRNC.equals(documentTypeCode)) {
```

Comment:
Get TotalPreTaxDollarAmount once instead of four times. Add here, then use below:
KualiDecimal totalPreTaxDollarAmount = getPurchasingAccountsPayableModuleService().getTotalPreTaxDollarAmount(document.getDocumentHeader().getDocumentNumber());